lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -0,0 +1,162 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+
+import grpc
+from lte.protos.s6a_service_pb2 import DeleteSubscriberRequest
+from lte.protos.s6a_service_pb2_grpc import S6aServiceStub
+from lte.protos.subscriberdb_pb2 import (ListSubscribersRequest,
+                                         LTESubscription, SubscriberData)
+from magma.common.grpc_client_manager import GRPCClientManager
+from magma.common.rpc_utils import grpc_async_wrapper
+from magma.common.sdwatchdog import SDWatchdogTask
+from magma.common.service import MagmaService
+from magma.common.service_registry import ServiceRegistry
+from magma.subscriberdb.store.sqlite import SqliteStore
+
+
+class SubscriberDBCloudClient(SDWatchdogTask):
+    """
+    SubscriberDBCloudClient for requesting subscribers from Orchestrator.
+
+    SubscriberDBCloudClient calls the Orchestrator's SubscriberDBCloud service
+    to fetch pages of subscribers stored in Orchestrator and updates the local
+    subscriberdb store with the changes.
+    """
+    DEFAULT_SUBSCRIBERDB_SYNC_INTERVAL = 180
+    DEFAULT_SUBSCRIBER_PAGE_SIZE = 250
+    SUBSCRIBERDB_REQUEST_TIMEOUT = 10
+
+    def __init__(self,
+                 service: MagmaService,
+                 store: SqliteStore,
+                 grpc_client_manager: GRPCClientManager):
+
+        sync_interval = service.config.get(
+            'subscriberdb_sync_interval',
+            self.DEFAULT_SUBSCRIBERDB_SYNC_INTERVAL)
+
+        super().__init__(
+            sync_interval,
+            service.loop
+        )
+        self._service = service
+        self._store = store
+
+        self._subscriber_page_size = service.config.get(
+            'subscriber_page_size', self.DEFAULT_SUBSCRIBER_PAGE_SIZE)
+
+        # grpc_client_manager to manage grpc client recycling
+        self._grpc_client_manager = grpc_client_manager
+
+    async def _run(self) -> None:
+        subscribers = await self._get_subscribers()
+        if subscribers is not None:
+            self._process_subscribers(subscribers)
+
+    async def _get_subscribers(self) -> [SubscriberData]:
```

Comment:
I wonder if we can add some metrics around this piece including total time to sync etc which we might want to use later in case we want more visibility around this one.